Automattic/jetpack
# WP REST API: Add VideoPress data to media endpoints

#### Changes proposed in this Pull Request:

This PR adds a new plugin that extends the media endpoints (`/wp/v2/media`) including VideoPress data so it can be used by Gutenberg (which retrieves data using `apiFetch` from `/wp/v2/media`). This plugin will be in sync with WordPress.com (D23321-code).

This is required by the VideoPress extension we're building for Gutenberg, since the VideoPress data is available currently only on `/rest/v1.1` endpoints that don't work with Jetpack sites.

#### Questions

I'm worried about the performance cost of this change. Do we have any way of checking it? Should I create an independent endpoint for retrieving the VideoPress data?

#### Testing instructions:
* Apply D23321-code, sandbox the API and create a new Jetpack site using JN: https://jurassic.ninja/create?jetpack-beta&branch=add/rest-api-media-videopress.
* Connect the Jetpack site to WordPress.com using a premium/business subscription.
* Go to Media > Library and upload an image and a video.
* Go to Jetpack > Settings and activate the VideoPress module (by enabling the "Enable high-speed, ad-free video player" option).
* Go to Media > Library and upload a new video.
* Perform a request to the media endpoint: `GET https://public-api.wordpress.com/wp/v2/sites/:site/media`.
* Check that the `jetpack_videopress_guid` property is included only on the video items (it should be an empty string for the video uploaded before activating VideoPress and the GUID value for the one uploaded after activating it).
* Deactivate the VideoPress module or remove the premium/business subscription.
* Perform again the request to the media endpoint.
* Make sure the `jetpack_videopress_guid` is still present on the video items.

#### Proposed changelog entry for your changes:

* WP REST API: Add VideoPress data to media endpoints
